File: week02/week02-3.cpp
class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        int N = s.length(); // C++ 的字串的長度
        int one = 0; // s 裡面 有幾個1呢?
        for(int i=0; i<N; i++){ // C/C++迴圈(有圓括號)
            if(s[i]=='1') one +=1; // if(判斷) 也有圓括號
        }
        //printf("N %d one %d", N, one); // 先印出來 有幾個1
        string ans;
        for (int i=0; i<one-1; i++) ans += '1'; // 有幾個1要放前面
        for (int i=0; i<N-one; i++) ans += '0'; // 有幾個0要放中間
        ans += '1'; //最後塞個1
        return ans; 
    }
};

File: week02/week02-3.py
class Solution:
    def maximumOddBinaryNumber(self, s: str) -> str:
        N = len(s)
        one = 0 # s裡面 有幾個1呢?
        for c in s: # 針對字串s的每個字母c
            if c=='1': one+=1 # 如果是'1'的話 one 就+1
        return '1'*(one-1) + '0'*(N-one) + '1'